lib/wash/exe/eval.js

```diff
@@ -0,0 +1,195 @@
+  });
+};
+
+/**
+ * Read-eval-print-loop.
+ *
+ * @return {Promise<*>} Resolves to the value of the final evaluation.
+ */
+Eval.prototype.readEvalPrintLoop = function() {
+  return this.readEvalPrint('').then(
+    function(value) {
+      if (this.exit) {
+        return Promise.resolve();
+      }
+      if (this.executeContext.isEphemeral('Ready'))
+        return this.readEvalPrintLoop();
+
+      return Promise.resolve(value);
+    }.bind(this)
+  ).catch(
+    function(value) {
+      return Promise.reject(value);
+    }.bind(this)
+  );
+};
+
+/**
+ * Read a single line of input, eval it, print the result or an error.
+ *
+ * @return {Promise<*>} Resolves to result of the evaluation.
+ */
+Eval.prototype.readEvalPrint = function(inputString) {
+  var input = inputString;
+  var promptString = this.promptString_;
+
+  if (inputString !== '') {
+    promptString = '';
+  }
+
+  return this.read(promptString).then(
+    function(result) {
+      if (result === null || result === 'exit') {
+          this.exit = true;
+        return Promise.resolve();
+      }
+
+      if (typeof result != 'string') {
+        return Promise.reject(new AxiomError.Runtime(
+            'Unexpected type from readline: ' + (typeof result)));
+      }
+
+      input += result;
+      return this.evaluate(input).then(
+        function(value) {
+          if (typeof value !== 'undefined' && value !== null) {
+            this.executeContext.stdout.write(value + '\n');
+          }
+          return Promise.resolve(value);
+        }.bind(this));
+    }.bind(this)
+  ).catch(
+    function(error) {
+      // We currently do not support multi line input in readline. This is a
+      // proxy to find if the error is a syntax error in code or the user has
+      // not finished typing.
+      // TODO(grv): Do a better detection of syntax error vs incomplete code.
+      if (error && error.message == 'Unexpected end of input') {
+        return this.readEvalPrint(input);
+      } else if (error) {
+        this.executeContext.stdout.write('Error: ' + error.message + '\n');
+      } else {
+        this.executeContext.stdout.write(
+          'Error in evaluating input:' + input + '\n');
+      }
+
+      return Promise.resolve();
+    }.bind(this)
+  );
+};
+
+/**
+ * @param {string} promptString
+ * @return {!Promise<*>}
+ */
+Eval.prototype.read = function(promptString) {
+  return washUtil.findExecutable(this.executeContext, 'readline').then(
+    function(result) {
+      return this.executeContext.call(
+          this.executeContext.fileSystemManager,
+          result.path,
+          { 'prompt-string': promptString,
+            'input-history': this.inputHistory
+          });
+    }.bind(this));
+};
+
+/**
+ * @param {string} input
+ * @return {Promise<*>}
+ */
+Eval.prototype.evaluate = function(input) {
+  console.log(input);
+  try {
+    var value = eval(input);
+    return Promise.resolve(value);
+  } catch (err) {
+    console.log(err);
```

Comment:
Nevermind, it is a console.log (I thought it was a stdout.write). Remove it then?